doc/feature/restAPI.md

```diff
@@ -0,0 +1,129 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented,
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+GET /registry returns the URIs for all registered objects, together with the
+supported REST operation (GET or PUT). Each objet can be queried for its schema
+using a GET operation on the URI of that object, suffixed by /schema.
+
+- Example:
+  - Get all URIs of registered objects:
+    * Request: ``` GET http://hostname:port/registry ```
+    * Response:
+```
+    {
+      { "namespace/foo": ["GET", "PUT"] },
+      { "namespace/bar": ["PUT"] }
+    }
+```
+  - Get JSON schema for a specific object:
+    * Request: ```GET http://hostname:port/namespace/foo/schema```
+    * Response:```{ "foo": { "name": string, "age": integer } }```
+
+## Examples
+Create an HTTP server and register serializables:
+```
+  void setupHTTPServer( const int argc, char** argv )
+  {
+    // Create HTTP server
+    _httpServer = zeq::http::Server::parse( argc, argv );
+    if( !_httpServer )
+      return;
+
+    // Register C++ objects
+    subscribers.push_back( _httpServer );
+    _httpServer.subscribe( camera, LUT, vrParams );
+    _httpServer.register( camera, LUT, vrParams, frame );
+  }
+```
+Query API:
+   * Request: ```GET http://hostname:port/zerobuf/render/camera/schema```
+   * Response: ```{ "camera" : { "origin": "string", "lookat": "up" } }```
```

Comment:
type of origin is more like zerobuf/render/vector3f. Please elaborate how nested ZeroBufs are exposed and handled in an example.